oracle/metrics/metrics.go

```diff
@@ -58,6 +67,11 @@ type Metrics interface {
 
 // OracleMetricsImpl is a Metrics implementation that does nothing.
 type OracleMetricsImpl struct {
+	promMetrics  *PromMetrics
+	statsdClient *statsd.Client
```

Comment:
Feels like this should be `statsd.ClientInterface` so that we can have a metrics impl that has nop statsd or has an actual client implementation.

Reply:
That would let us avoid all of the if branching around nil stasd clients in the method impls as well.

Reply:
In my mind the nil check makes it clear this can be toggled on or off, but I see where this could be bad if we ever forget to check if it is nil and try to call that method

Reply:
So will change